### PR TITLE
Fixes StaticCanvas.onBeforeScaleRotate not triggering when canvas is zoomed

### LINKED ISSUE #4746: canvas.onBeforeScaleRotate not triggering while canvas is zoomed
## Version
2.0.2

## Test Case
http://jsbin.com/gunatuyusa/edit?html,js,output

## Steps to reproduce
Try to scale, notice fill change.
Double click canvas, canvas is now zoomed.
Try to scale using bottom right corner, fill is not changed.

## Expected Behavior
On canvas zoom, the callback should still be triggered

## Actual Behavior
Callback not fired.

---

Narrowed down the issue to code:

src/mixins/canvas_events.mixin.js:576

```
      // determine if it's a drag or rotate case
      if (target._findTargetCorner(this.getPointer(e))) {
        this.onBeforeScaleRotate(target);
      }
```

It should be `getPointer(e, true)` ? I haven't tested it.
I do not have the development flow set up now. Will spend time when I have it.

### PR DESCRIPTION
**Pull request by scriptspry**
Fixes #4746

### CODE REVIEW

**Comment by asturur** [comment]
Cool!
You should remove the changes to dist before pushing

try with

`git checkout master dist`

and then commit/push again

**Comment by asturur** [comment]
Would be nice to add a small test

**Comment by asturur** [comment]
If there are no test examples with event generation i ll help, but i think there is something around.

**Comment by scriptspry** [comment]
Took me an hour to figure out where, how and what to test. Let me know if there are any changes. Will do it tomorrow. First time took me a lot of time. Shouldn't take long next time.

**Comment by scriptspry** [comment]
I have another doubt: shouldn't onBeforeScaleRotate be member of Canvas instead of StaticCanvas?

**Comment by asturur** [comment]
definetely. but do not worry to change it now, is a potentially dangerous change for which may be test missing.

File: test/unit/canvas.js [hunk 1]
```diff
@@ -2337,6 +2337,56 @@
...
+        which: 1
+      };
+      canvas._beforeTransform(e, rect);
+    }
+    assert.equal(counter, corners.length, '_beforeTransform should trigger onBeforeScaleRotate when canvas is zoomed');
+    assert.equal(t, rect, 'onBeforeScaleRotate should receive correct target when canvas is zoomed');
+
+    canvas.zoomToPoint({ x: 0, y: 0 }, 1);
+    canvas.onBeforeScaleRotate = _onBeforeScaleRotate;
+  });
```
Comment (asturur) [inline test/unit/canvas.js]: perfect test.

**Comment by asturur** [comment]
Is good, thanks for the test, is extensive and well written.

**Pull request merged**